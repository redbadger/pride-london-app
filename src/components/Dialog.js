// @flow
import React from "react";
import type { Node } from "react";
import {
  Modal,
  StyleSheet,
  TouchableWithoutFeedback,
  View
} from "react-native";
import Button from "./ButtonPrimary";
import {
  dialogBackdropColor,
  dialogBgColor,
  dialogHeaderDividerColor
} from "../constants/colors";

type Props = {
  applyButtonText: string,
  applyButtonLabel: string,
  applyButtonDisabled?: boolean,
  children: Node,
  headerLeft?: Node,
  headerRight?: Node,
  onApply: Function,
  onCancel?: Function,
  title: Node,
  visible: boolean
};

const Dialog = ({
  applyButtonText,
  applyButtonLabel,
  applyButtonDisabled,
  children,
  headerLeft,
  headerRight,
  onApply,
  onCancel,
  title,
  visible
}: Props) => (
  <Modal
    animationType="fade"
    onRequestClose={onCancel}
    transparent
    visible={visible}
  >
    <TouchableWithoutFeedback
      style={styles.backdrop}
      onPress={onCancel}
      accessible={false}
    >
      <View style={styles.container}>
        <TouchableWithoutFeedback accessible={false}>
          <View>
            <View style={styles.content}>
              <View style={styles.header}>
                <View style={styles.headerSideLeft}>{headerLeft}</View>
                <View style={styles.headerSideMiddle}>{title}</View>
                <View style={styles.headerSideRight}>{headerRight}</View>
              </View>
              {children}
            </View>
            <View style={styles.applyButtonContainer}>
              <Button
                disabled={applyButtonDisabled}
                onPress={onApply}
                accessibilityLabel={applyButtonLabel}
              >
                {applyButtonText}
              </Button>
            </View>
          </View>
        </TouchableWithoutFeedback>
      </View>
    </TouchableWithoutFeedback>
  </Modal>
);

Dialog.defaultProps = {
  headerLeft: undefined,
  headerRight: undefined,
  onCancel: () => {},
  applyButtonDisabled: false
};

const styles = StyleSheet.create({
  backdrop: {
    flex: 1
  },
  container: {
    flex: 1,
    justifyContent: "center",
    padding: 16,
    backgroundColor: dialogBackdropColor
  },
  content: {
    backgroundColor: dialogBgColor,
    borderRadius: 4,
    paddingBottom: 8
  },
  header: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    height: 40,
    paddingHorizontal: 16,
    borderBottomWidth: 1,
    borderBottomColor: dialogHeaderDividerColor
  },
  headerSideLeft: {
    alignItems: "flex-start"
  },
  headerSideMiddle: {
    alignItems: "center"
  },
  headerSideRight: {
    alignItems: "flex-end"
  },
  applyButtonContainer: {
    marginTop: 8
  }
});

export default Dialog;
