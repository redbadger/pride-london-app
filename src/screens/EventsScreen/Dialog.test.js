// @flow
import React from "react";
import { shallow } from "enzyme";
import Dialog from "./Dialog";

it("renders correctly", () => {
  const output = shallow(
    <Dialog
      applyButtonText="Bye"
      applyButtonLabel="Good bye"
      onApply={() => {}}
      title="Hello World"
      visible
    >
      Lorem Ipsum is placeholder text commonly used...
    </Dialog>
  );
  expect(output).toMatchSnapshot();
});
