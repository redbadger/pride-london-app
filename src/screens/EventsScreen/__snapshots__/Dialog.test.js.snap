// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Component
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <TouchableWithoutFeedback
    accessible={false}
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Component
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.7)",
          "flex": 1,
          "justifyContent": "center",
          "padding": 16,
        }
      }
    >
      <TouchableWithoutFeedback
        accessible={false}
      >
        <Component>
          <Component
            style={
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 4,
                "paddingBottom": 8,
              }
            }
          >
            <Component
              style={
                Object {
                  "alignItems": "center",
                  "borderBottomColor": "#F2F2F2",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "height": 40,
                  "justifyContent": "space-between",
                  "paddingHorizontal": 16,
                }
              }
            >
              <Component
                style={
                  Object {
                    "alignItems": "flex-start",
                  }
                }
              />
              <Component
                style={
                  Object {
                    "alignItems": "center",
                  }
                }
              >
                Hello World
              </Component>
              <Component
                style={
                  Object {
                    "alignItems": "flex-end",
                  }
                }
              />
            </Component>
            Lorem Ipsum is placeholder text commonly used...
          </Component>
          <Component
            style={
              Object {
                "marginTop": 8,
              }
            }
          >
            <Button
              accessibilityLabel="Good bye"
              disabled={false}
              onPress={[Function]}
            >
              Bye
            </Button>
          </Component>
        </Component>
      </TouchableWithoutFeedback>
    </Component>
  </TouchableWithoutFeedback>
</Component>
`;
